### DISCUSSION TITLE
Integrating Radashi.js with GPT

### DISCUSSION THREAD

**Discussion by MarlonPassos-git** [radashi-org/radashi#216] [Ideas]
Daily, I use some personalized versions of ChatGPT, like:

https://chatgpt.com/g/g-0xuCnkXju-vim-wizard
https://chatgpt.com/g/g-erKUuASLu-zustand-state-management

These customizations help me quite a lot. At some point, it might be interesting to create a similar version for Radashi that could be powered by our documentation. I imagine this could be useful for people to refactor code, such as "refactor this code using Radashi utilities" or by asking whether Radashi has a utility for X, Y, Z.

I know that for users without a paid plan, there are input limitations, but it’s still possible to experiment. And since the idea is to promote Radashi in various places, why not in ChatGPT?

Of course, there are other platforms, but thinking about practicality, if someone already has access to the Plus version, we could implement it.

**Comment by aleclarson** [comment]
Yeah that'd be cool. I had the same thought. Apparently, you can't publish a custom GPT there unless you pay monthly, but the GPTs are available even to free users these days, I heard.

**Reply by MarlonPassos-git** [reply]
Yes, it is possible to use it for free with certain limitations. Based on my experience, you can send around 20 messages every 4 hours, which is generally fine.

https://openai.com/index/gpt-4o-and-more-tools-to-chatgpt-free/

I currently have access to GPT Plus. So initially, I could create a custom GPT there, and in the future, if we secure sponsorship, we could have a dedicated account for Radashi.

**Reply by aleclarson** [reply]
Alright, sounds good. Are there other popular libraries that have done this? It'd be nice to see what they've done. Otherwise, do you know what it would take? Do we need to generate [embeddings](https://platform.openai.com/docs/guides/embeddings) or is it as simple as feeding in the documentation?

**Reply by MarlonPassos-git** [reply]
Some that I have already used:
- React: https://chatgpt.com/g/g-w2D4rqH0d-react
- Zustand: https://chatgpt.com/g/g-erKUuASLu-zustand-state-management
- Radix UI: https://chatgpt.com/g/g-QDDaEAn9f-radix-ui-ranger-ai
- FlowBite: https://chatgpt.com/g/g-y7yC35HB9-flowbite-gpt
- Material UI: https://chatgpt.com/g/g-QlHCQzA49-react-material-ui-expert
- Next.js: https://chatgpt.com/g/g-lAbjRB9qC-next-js-nextjs

Now how it works, I’ll have to study it a bit.

But I’ll get to that after I manage to resolve the issues with the dependency updates and the search terms.

**Comment by MarlonPassos-git** [comment]
It looks like github copilot has changed things a bit

![image](https://github.com/user-attachments/assets/8552ceba-8375-4ee3-b248-6ac9eca4356b)
![image](https://github.com/user-attachments/assets/8ca1bf12-2c7d-4351-ad07-a64ed156ff2a)
![image](https://github.com/user-attachments/assets/650fc8c3-9336-45e1-8ced-2d1e65f6413c)


It has some problems, but I found it interesting that with one click of a button it analyzes the entire repository

**Reply by MarlonPassos-git** [reply]
And they have plugins https://github.com/marketplace?type=apps&copilot_app=true that can integrate with VsCode
